### PR TITLE
refactor(monorepo): relocate deckgl to Superset

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

This PR move [Deckgl](https://github.com/apache-superset/superset-ui-plugins-deckgl) to plugins directory
1. move legacy-preset-chart-deckgl to plugins directory
2. move storybook to `packages/superset-ui-demo/stories/plugins`
3. add `tsconfig.json` into package
4. refactor storybook of Deckgl
5. refactor `src/factory.jsx` to `typescript`
6. remove `legacy-preset-chart-deckgl` from `jest.config.js`(in order to mapping this plugin for test)
7. change `controlPanel` to `typescript`
8. fix lint/type/test
9. add Apache License

### TESTING INSTRUCTIONS

1. pass all steps of CI
2. manually test in local environment

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by stephenLYZ** [comment]
LGTM

File: superset-frontend/packages/superset-ui-demo/storybook/stories/plugins/legacy-preset-chart-deckgl/Arc/Stories.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * under the License.
+ */
+
+/* eslint-disable sort-keys */
+/* eslint-disable no-magic-numbers */
+import React from 'react';
+import { SuperChart } from '@superset-ui/core';
+import { ArcChartPlugin } from '@superset-ui/legacy-preset-chart-deckgl';
+import payload from './payload';
+import dummyDatasource from '../../../../shared/dummyDatasource';
```
Comment (kgabryje) [inline superset-frontend/packages/superset-ui-demo/storybook/stories/plugins/legacy-preset-chart-deckgl/Arc/Stories.tsx:26]: NIT: can we simplify this import? Maybe by setting a base path in `tsconfig.json` to `storybook`? Then we could import like `import dummyDatasource from 'storybook/shared/dummyDatasource';` WDYT?

Comment (kgabryje) [reply superset-frontend/packages/superset-ui-demo/storybook/stories/plugins/legacy-preset-chart-deckgl/Arc/Stories.tsx:26]: (comment applies to other files below that import dummyDatasource like that)

Comment (zhaoyongjie) [reply superset-frontend/packages/superset-ui-demo/storybook/stories/plugins/legacy-preset-chart-deckgl/Arc/Stories.tsx:26]: The `absolute import` depends on how to locate the `storybook/shared` module. (Maybe needed module alias in somewhere).

so `relative import` is straightforward here.

**Review by kgabryje — APPROVED** [review]
LGTM

**Pull request merged**